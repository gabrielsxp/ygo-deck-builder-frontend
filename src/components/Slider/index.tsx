import * as S from './styles'
import SlickSlider, { Settings } from 'react-slick'

export type SliderSettings = Settings

export type SliderProps = {
  children: React.ReactNode
  settings: SliderSettings
  color?: 'light' | 'dark' | 'medium'
}

const Slider = ({ children, settings, color = 'light' }: SliderProps) => (
  <S.Wrapper color={color}>
    <SlickSlider {...settings}>{children}</SlickSlider>
  </S.Wrapper>
)

export default Slider
