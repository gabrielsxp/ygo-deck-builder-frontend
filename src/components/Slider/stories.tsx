import { Story, Meta } from '@storybook/react/types-6-0'
import { Settings } from 'react-slick'
import Slider, { SliderSettings, SliderProps } from '.'
import styled from 'styled-components'

export default {
  title: 'Slider',
  component: Slider
} as Meta

const settings: Settings = {
  dots: true,
  infinite: true,
  speed: 500,
  slidesToShow: 3,
  slidesToScroll: 1
}
const verticalSettings: Settings = {
  vertical: true,
  verticalSwiping: true,
  dots: true,
  infinite: true,
  speed: 500,
  slidesToShow: 1,
  slidesToScroll: 1
}

const Slide = styled.div`
  background-color: lightgrey;
  padding: 20rem 0;
  display: flex;
  justify-content: center;
  align-items: center;
  border: 0.1rem solid red;
  font-weight: bold;
  font-size: 5rem;
  text-align: center;
  border-radius: 0.5rem;
`

export const Horizontal: Story<SliderProps> = () => (
  <Slider settings={settings}>
    {Array.from(new Array(5)).map((_, index) => (
      <Slide key={index}>{index + 1}</Slide>
    ))}
  </Slider>
)
export const Vertical: Story<SliderProps> = () => (
  <Slider settings={verticalSettings}>
    {Array.from(new Array(5)).map((_, index) => (
      <Slide key={index}>{index + 1}</Slide>
    ))}
  </Slider>
)
