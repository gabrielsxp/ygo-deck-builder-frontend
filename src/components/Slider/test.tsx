import 'match-media-mock'
import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/helpers'

import Slider from '.'

describe('<Slider />', () => {
  it('should render children as slider item', () => {
    const { container } = renderWithTheme(
      <Slider settings={{ infinite: false, slidesToShow: 2 }}>
        <p>Slide 1</p>
        <p>Slide 2</p>
      </Slider>
    )
    expect(
      screen.getByText(/slide 1/i).parentElement?.parentElement
    ).toHaveClass('slick-slide')
    expect(
      screen.getByText(/slide 2/i).parentElement?.parentElement
    ).toHaveClass('slick-slide')
    expect(container.firstChild).toMatchSnapshot()
  })
})
