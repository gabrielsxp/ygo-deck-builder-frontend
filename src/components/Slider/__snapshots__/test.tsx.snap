// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> should render children as slider item 1`] = `
.c0 {
  width: 100%;
}

.c0 .slick-slider {
  position: relative;
  display: block;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-touch-callout: none;
  -khtml-user-select: none;
  -ms-touch-action: pan-y;
  touch-action: pan-y;
  -webkit-tap-highlight-color: transparent;
}

.c0 .slick-list {
  position: relative;
  display: block;
  overflow: hidden;
  margin: 0;
  padding: 0;
}

.c0 .slick-list:focus {
  outline: none;
}

.c0 .slick-list.dragging {
  cursor: pointer;
  cursor: hand;
}

.c0 .slick-slider .slick-track,
.c0 .slick-slider .slick-list {
  -webkit-transform: translate3d(0,0,0);
  -moz-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  -o-transform: translate3d(0,0,0);
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
}

.c0 .slick-track {
  position: relative;
  top: 0;
  left: 0;
  display: block;
}

.c0 .slick-track:before,
.c0 .slick-track:after {
  display: table;
  content: '';
}

.c0 .slick-track:after {
  clear: both;
}

.c0 .slick-loading .slick-track {
  visibility: hidden;
}

.c0 .slick-slide {
  display: none;
  float: left;
  height: 100%;
  min-height: 1px;
}

.c0 [dir='rtl'] .slick-slide {
  float: right;
}

.c0 .slick-slide img {
  display: block;
}

.c0 .slick-slide.slick-loading img {
  display: none;
}

.c0 .slick-slide.dragging img {
  pointer-events: none;
}

.c0 .slick-initialized .slick-slide {
  display: block;
}

.c0 .slick-loading .slick-slide {
  visibility: hidden;
}

.c0 .slick-vertical .slick-slide {
  display: block;
  height: auto;
  border: 1px solid transparent;
}

.c0 .slick-arrow.slick-hidden {
  display: none;
}

.c0 .slick-prev,
.c0 .slick-next {
  font-size: 0;
  line-height: 0;
  position: absolute;
  top: 50%;
  display: block;
  width: 4rem;
  height: 4rem;
  padding: 0;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
  cursor: pointer;
  color: #030517;
  border: none;
  outline: none;
  background: transparent;
}

.c0 .slick-prev:hover,
.c0 .slick-prev:focus,
.c0 .slick-next:hover,
.c0 .slick-next:focus {
  color: rgba(3,5,23,0.7);
  outline: none;
  background: transparent;
}

.c0 .slick-prev:hover:before,
.c0 .slick-prev:focus:before,
.c0 .slick-next:hover:before,
.c0 .slick-next:focus:before {
  opacity: 1;
}

.c0 .slick-prev.slick-disabled:before,
.c0 .slick-next.slick-disabled:before {
  opacity: 0.25;
}

.c0 .slick-prev:before,
.c0 .slick-next:before {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c0 .slick-prev {
  left: -75px;
}

.c0 [dir='rtl'] .slick-prev {
  right: -75px;
  left: auto;
}

.c0 .slick-next {
  right: -75px;
}

.c0 [dir='rtl'] .slick-next {
  right: auto;
  left: -75px;
}

.c0 .slick-dotted.slick-slider {
  margin-bottom: 30px;
}

.c0 .slick-dots {
  position: absolute;
  bottom: -40px;
  display: block;
  width: 100%;
  padding: 0;
  margin: 0;
  list-style: none;
  text-align: center;
}

.c0 .slick-dots li {
  position: relative;
  display: inline-block;
  width: 20px;
  height: 20px;
  margin: 0 5px;
  padding: 0;
  cursor: pointer;
}

.c0 .slick-dots li button {
  font-size: 0;
  line-height: 0;
  display: block;
  width: 20px;
  height: 20px;
  padding: 5px;
  cursor: pointer;
  color: transparent;
  border: 0;
  outline: none;
  background: transparent;
}

.c0 .slick-dots li button:hover,
.c0 .slick-dots li button:focus {
  outline: none;
}

.c0 .slick-dots li button:hover:before,
.c0 .slick-dots li button:focus:before {
  opacity: 1;
}

.c0 .slick-dots li button:before {
  font-family: 'slick';
  font-size: 3rem;
  line-height: 20px;
  position: absolute;
  top: 0;
  left: 0;
  width: 20px;
  height: 20px;
  content: 'â€¢';
  text-align: center;
  opacity: 0.25;
  color: #d5057f;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c0 .slick-dots li.slick-active button:before {
  opacity: 1;
  color: #030517;
}

<section
  class="c0"
  color="light"
>
  <div
    class="slick-slider slick-initialized"
  >
    
    <div
      class="slick-list"
    >
      <div
        class="slick-track"
        style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
      >
        <div
          aria-hidden="false"
          class="slick-slide slick-active slick-current"
          data-index="0"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <p
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              Slide 1
            </p>
          </div>
        </div>
        <div
          aria-hidden="false"
          class="slick-slide slick-active"
          data-index="1"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <p
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              Slide 2
            </p>
          </div>
        </div>
      </div>
    </div>
    
    
  </div>
</section>
`;
